manifests/allow_client_on_server.pp

```diff
@@ -0,0 +1,56 @@
+# @summary Enable backup client to access backup server
+#
+# @param hostname
+#   fqdn hostname of backup client.
+#
+# @param ip
+#   ip of backup client.
+#
+# @param key
+#   The public ssh key itself; generally a long string...
+#
+# @param key_type
+#   The encryption type used in the public ssh key.
+#
+# @example
+#   profile_backup::allow_client_on_server { 'allow host hostname access to backup servers':
+#     'hostname' => String,
+#     'ip' => String,
+#     'key' => String,
+#     'key_type' => String,
+#   }
+define profile_backup::allow_client_on_server (
+  String $hostname,
+  String $ip,
+  String $key,
+  String $key_type,
```

Comment:
suggest remove this parameter, allow putting the entire ssh_key into "ssh_key_pub"
then, ssh_key_pub can be broken into it's parts in the puppet code

Reply:
We are adding several complex `ssh_authorized_key` `options` and `name`/comment to each authorized key entry we're adding. 
I don't think it makes sense to combine them because then I will not be able to utilize `ssh_authorized_key` which validates each parameter that it uses.

Reply:
Just noting here we talked in person and I agree using the same format as used elsewhere (ie: ssh_authorized_key) makes the most sense.